spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -482,14 +489,14 @@ else if (result instanceof Message) {
 					iterableOfMessages = iterator.next() instanceof Message;
 				}
 				if (iterableOfMessages || this.splitIterables) {
-					((Iterable<V>) result).forEach(v -> {
+					for (V v : (Iterable<V>) result) {
 						if (v instanceof Message) {
 							this.replyTemplate.send((Message<?>) v);
 						}
 						else {
 							this.replyTemplate.send(topic, v);
 						}
-					});
+					}
```

Comment:
lambda has to be switched to loop for `topic` is not effective final after logic change